aptos-move/aptos-gas/src/gas_meter.rs

```diff
@@ -145,18 +145,33 @@ impl InitialGasSchedule for AptosGasParameters {
 
 /// The official gas meter used inside the Aptos VM.
 /// It maintains an internal gas counter, measured in internal gas units, and carries an environment
-/// consisting all the gas parameters, which it can lookup when performing gas calcuations.
+/// consisting all the gas parameters, which it can lookup when performing gas calculations.
 pub struct AptosGasMeter {
+    feature_version: u64,
     gas_params: AptosGasParameters,
+    storage_gas_params: Option<StorageGasParameters>,
     balance: InternalGas,
 }
 
 impl AptosGasMeter {
-    pub fn new(gas_params: AptosGasParameters, balance: impl Into<Gas>) -> Self {
+    pub fn new(
+        gas_feature_version: u64,
+        gas_params: AptosGasParameters,
+        storage_gas_params: Option<StorageGasParameters>,
+        balance: impl Into<Gas>,
+    ) -> Self {
+        assert!(
+            (gas_feature_version == 0 && storage_gas_params.is_none())
+                || storage_gas_params.is_some(),
```

Comment:
This is equivalent to `gas_feature_version == 0 || storage_gas_params.is_some()`, is that what you meant? 

-- because I thought you wanted something more restrictive, like `gas_feature_version == 0 && storage_gas_params.is_none() || gas_feature_version != 0 && storage_gas_parameter.is_some()`

Reply:
Good catch. I think I made a mistake when simplifying the boolean expression.

Reply:
Will put up a fix now